src/app/shared/components/burger-menu/burger-menu.component.ts

```diff
@@ -0,0 +1,10 @@
+import { Component, Input } from "@angular/core";
+
+@Component({
+    selector: "app-burger-menu",
+    templateUrl: "./burger-menu.component.html",
+    styleUrls: ["./burger-menu.component.scss"]
+})
+export class BurgerMenuComponent {
+    @Input() onBurgerMenuClick: () => void = () => {};
```

Comment:
use @Output instead. That way, instead of passing down the click handler, we can handle the callback in the parent component